bundle/phases/deploy.go

```diff
@@ -155,6 +155,7 @@ func Deploy(outputHandler sync.OutputHandler) bundle.Mutator {
 				mutator.ValidateGitDetails(),
 				artifacts.CleanUp(),
 				libraries.ExpandGlobReferences(),
+				libraries.CheckForSameNameLibraries(),
```

Comment:
Could you add a comment there saying that 
- this must be run after build
- this must be run after globs are expanded.